### PR TITLE
Upgrade to typeorm 0.3.7

### PR DESCRIPTION
**Pull request by GP4cK**
This PR upgrades to typeorm 0.3.7 (and NestJS v9 in the query-typeorm package).

There are 2 things which I haven't committed and wanted to ask your opinion:

1. This repo uses Prettier to format the code. Because I didn't have the [VSCode extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) installed, I always had conflicts when saving file because the default TS formatter / ESLint didn't agree with Prettier. Adding this file will make sure that the Prettier extension is used.
<details>
<summary>.vscode/settings.json</summary>

```json
{
  "editor.defaultFormatter": "esbenp.prettier-vscode"
}
```
</details>

3. For some reason, by default, I wasn't to debug the tests in `packages/query-typeorm/__tests__/services/typeorm-query.service.spec.ts` with breakpoints which made it pretty difficult to troubleshoot failing tests. I created an issue here but a workaround I found was to add this file.
<details>
<summary>.vscode/launch.json</summary>

```json
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug Jest Tests",
      "type": "node",
      "request": "launch",
      "runtimeArgs": [
        "--inspect-brk", 
        "${workspaceRoot}/node_modules/.bin/jest", 
        "--runInBand",
        "-t",
        "set the relation to null", // replace with name of your test
      ],
      "console": "integratedTerminal",
      "internalConsoleOptions": "neverOpen"
    }
  ]
}
```
</details>

Finally, typeorm 0.3.x marked a lot of methods as deprecated but they still work (getConnection() etc.). I've only changed the minimum for the code to run and the tests to pass. When this gets merged, I can work on updating the deprecated code.

### CODE REVIEW

File: packages/query-typeorm/src/services/typeorm-query.service.ts [hunk 1]
```diff
@@ -272,7 +273,7 @@ export class TypeOrmQueryService<Entity>
       const distinctRecords = await builder.addSelect(`${builder.alias}.id`).getRawMany();
 
       const ids: unknown[] = distinctRecords.map(({ id }) => id as unknown);
-      const idsFilter = { id: { in: ids } } as Filter<Entity>;
+      const idsFilter = { id: { in: ids } } as unknown as Filter<Entity>;
```
Comment (TriPSs) [inline packages/query-typeorm/src/services/typeorm-query.service.ts:276]: Any specific reason why dis goes to unknown first?

**Comment by TriPSs** [comment]
Thanks for the PR! In #26 I updated a lot of other deps but not Typeorm yet, so this is a nice one!

File: packages/query-typeorm/src/services/typeorm-query.service.ts (hunk 1 above)
Comment (GP4cK) [reply packages/query-typeorm/src/services/typeorm-query.service.ts:276]: And that's even on the master branch.

**Comment by GP4cK** [comment]
> Thanks for the PR! In #26 I updated a lot of other deps but not Typeorm yet, so this is a nice one!

Yes I saw but my browser crashes when I try to open the PR to see your changes because there are too many modifications I think 🥲

**Comment by TriPSs** [comment]
FYI: I have the following plan:
1. Fix last things in my pr
2. Merge your PR into my PR
3. Test everything in my application to be even more sure that everything still works
4. Profit 🎉

**Comment by TriPSs** [comment]
Not sure how I committed into your branch but this is now part of #26, will release it as a alpha release soon so you can start using it and it can be fully tested.

**Comment by GP4cK** [comment]
> Not sure how I committed into your branch [...]

Uh yes. Probably because I ticked the box "Allow edits and access to secrets by maintainers" and I see from the git history that you merged the branch "improvements" into my branch.

So I'll close this and we will track changes on #26 then.

**Comment by TriPSs** [comment]
This is now released in `v2.0.0-alpha.1`

**Comment by GP4cK** [comment]
I don't think I can upgrade yet:
image

Maybe I'll just wait for your PR to be merged.

**Comment by TriPSs** [comment]
I tried upgrading that one but all the tests kept failing, will investigate soon to try to resolve that.

Currently also trying out the v2 alpha in a project of mine to see if everything runs ok.

**Comment by TriPSs** [comment]
@GP4cK did you still have issues with `@nestjs/common`? As it's a peer dep it should not block the usage.

The project I'm testing the v2 alpha in had no issues at all atm so we can update that peer to be v9 of nestjs without it being a issue.

**Comment by GP4cK** [comment]
@TriPSs yes `npm i` still outputs this error message. Are you using `npm i --force`? Do you mind sharing the content of your package.json dependencies?

**Comment by TriPSs** [comment]
@GP4cK im using yarn, will update the peer deps and release it.

**Pull request closed without merging**